client/src/pages/Login.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react";
+
+import Input from '@material-ui/core/Input';
+import Button from '@material-ui/core/Button';
+
+function Login() {
+
+    const [validationState, setValidationState] = useState(false);
+
+    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    function validateEmail(email) {
+        return re.test(String(email));
+    }
+
+    const handleInput = event => {
+        if (validateEmail(event.target.value)) {
+            setValidationState(true);
+        }
+        else {
+            setValidationState(false);
+        }
+    };
```

Comment:
This seems fine. Obviously here you also want to change the state of your actual input as well. so you'll need a state for your form `const [email, setEmail] = useState("");`, then in handleInput you do `setEmail(event.target.value);`